File: app/src/main/java/de/chrisward/theyworkforyou/viewmodel/MPViewModel.java
package de.chrisward.theyworkforyou.viewmodel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LifecycleOwner;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

import de.chrisward.theyworkforyou.database.MPStore;
import de.chrisward.theyworkforyou.database.TheyWorkForYouDatabase;
import de.chrisward.theyworkforyou.model.MP;
import de.chrisward.theyworkforyou.repository.MPRepository;
import de.chrisward.theyworkforyou.wrapper.Resource;

public class MPViewModel extends AndroidViewModel {
    public final LiveData<List<MP>> mpList;

    public MPViewModel(@NonNull Application application) {
        super(application);
        mpList = TheyWorkForYouDatabase.get(application).mpStore().selectAllMPs();
    }

    public void refreshMPs(LifecycleOwner lifecycleOwner) {
        MPRepository.getInstance(getApplication()).refreshMPs().observe(
                lifecycleOwner,
                mps -> {
                    if (mps == null) {
                        return;
                    }

                    switch (mps.status) {
                        case Resource.SUCCESS: {
                            final MPStore store = TheyWorkForYouDatabase.get(getApplication()).mpStore();

                            new Thread() {
                                @Override
                                public void run() {
                                    if (mps.data == null) {
                                        return;
                                    }

                                    TheyWorkForYouDatabase.get(getApplication()).clearAllTables();

                                    for (MP mp : mps.data) {
                                        store.insert(mp);
                                    }
                                }
                            }.start();
                            break;
                        }
                        case Resource.ERROR: {
                            //ERROR HANDLING HERE
                        }
                    }
                }
        );
    }
}

File: app/src/main/java/de/chrisward/theyworkforyou/viewmodel/MPViewModel.kt
package de.chrisward.theyworkforyou.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import de.chrisward.theyworkforyou.database.MPStore
import de.chrisward.theyworkforyou.database.TheyWorkForYouDatabase
import de.chrisward.theyworkforyou.model.MP
import de.chrisward.theyworkforyou.repository.MPRepository
import de.chrisward.theyworkforyou.wrapper.Resource
import javax.inject.Inject

class MPViewModel
@Inject constructor(
    application: Application,
    val mpRepository: MPRepository,
    val theyWorkForYouDatabase: TheyWorkForYouDatabase,
    val mpStore: MPStore
) : AndroidViewModel(application) {
    val mpList: LiveData<List<MP>>

    init {
        mpList = mpStore.selectAllMPs()
    }

    fun refreshMPs(lifecycleOwner: LifecycleOwner) {
        mpRepository
            .refreshMPs()
            .observe(
                lifecycleOwner,
                Observer<Resource<List<MP>>> { mps ->
                    when (mps!!.status) {
                        Resource.Status.SUCCESS -> {
                            object : Thread() {
                                override fun run() {
                                    if (mps.data == null) {
                                        return
                                    }

                                    theyWorkForYouDatabase.clearAllTables()

                                    for (mp in mps.data.orEmpty()) {
                                        mpStore.insert(mp)
                                    }
                                }
                            }.start()
                        }
                        Resource.Status.ERROR -> {
                            //ERROR HANDLING HERE
                        }
                        Resource.Status.LOADING -> {

                        }
                    }
                }
        )
    }
}
